receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -64,6 +72,87 @@ func (prw *prometheusRemoteWriteReceiver) Shutdown(ctx context.Context) error {
 	return prw.server.Shutdown(ctx)
 }
 
-func (prw *prometheusRemoteWriteReceiver) handlePRW(_ http.ResponseWriter, _ *http.Request) {
-	fmt.Println("handleWrite called")
+func (prw *prometheusRemoteWriteReceiver) handlePRW(w http.ResponseWriter, req *http.Request) {
+	contentType := req.Header.Get("Content-Type")
+	if contentType == "" {
+		prw.settings.Logger.Warn("message received without Content-Type header, rejecting")
+		http.Error(w, "Content-Type header is required", http.StatusUnsupportedMediaType)
+		return
+	}
+
+	msgType, err := prw.parseProto(contentType)
+	if err != nil {
+		prw.settings.Logger.Warn("Error decoding remote-write request", zapcore.Field{Key: "error", Type: zapcore.ErrorType, Interface: err})
+		http.Error(w, err.Error(), http.StatusUnsupportedMediaType)
+		return
+	}
+	if msgType != promConfig.RemoteWriteProtoMsgV2 {
+		prw.settings.Logger.Warn("message received with unsupported proto version, rejecting")
+		http.Error(w, "Unsupported proto version", http.StatusUnsupportedMediaType)
+		return
+	}
+
+	// After parsing the content-type header, the next step would be to handle content-encoding.
+	// Luckly confighttp's Server has middleware that already decompress the request body for us.
+
+	body, err := io.ReadAll(req.Body)
+	if err != nil {
+		prw.settings.Logger.Warn("Error decoding remote write request", zapcore.Field{Key: "error", Type: zapcore.ErrorType, Interface: err})
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var prw2Req writev2.Request
+	if err = proto.Unmarshal(body, &prw2Req); err != nil {
+		prw.settings.Logger.Warn("Error decoding remote write request", zapcore.Field{Key: "error", Type: zapcore.ErrorType, Interface: err})
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	_, stats, errHTTPCode, err := prw.translateV2(req.Context(), &prw2Req)
```

Comment:
nit: it feels like this should just pass the `http.ResponseWriter` to the translate function, rather than making it return so many things.

Reply:
Ah, I see.  We support partial success.  In that case, maybe just return the error and not the status code, since we can assume a failure to translate will always be bad input data.

Reply:
Yes, the initial idea was differentiating partial and full writes, but on second thought it's not required that we implement partial writes[[1](https://prometheus.io/docs/specs/remote_write_spec_2_0/#partial-write)]. I guess in our context it doesn't make sense to support that because we're not idempotent[[2](https://prometheus.io/docs/specs/remote_write_spec_2_0/#retries-on-partial-writes)].

Let's forget about different responses, any error = no writes!

Reply:
Following https://prometheus.io/docs/specs/remote_write_spec_2_0/#invalid-samples, I'm now returning 400 on any translation that returns errors